internal/configs/transportserver.go

```diff
@@ -172,13 +172,15 @@ func generateStreamUpstream(upstream *conf_v1alpha1.Upstream, upstreamNamer *ups
 
 	name := upstreamNamer.GetNameForUpstream(upstream.Name)
 	maxFails := generateIntFromPointer(upstream.MaxFails, 1)
+	maxConnections := generateIntFromPointer(upstream.MaxConnections, 0)
 	failTimeout := generateTimeWithDefault(upstream.FailTimeout, "10s")
 
 	for _, e := range endpoints {
 		s := version2.StreamUpstreamServer{
-			Address:     e,
-			MaxFails:    maxFails,
-			FailTimeout: failTimeout,
+			Address:        e,
+			MaxFails:       maxFails,
+			FailTimeout:    failTimeout,
+			MaxConnections: maxConnections,
```

Comment:
@soneillf5 respective unit tests in https://github.com/nginxinc/kubernetes-ingress/blob/feature-max-connections-transportserver-upstream-server/internal/configs/transportserver_test.go mainly around value validation and errors on wrong config.

Reply:
I've added a little unit test. I'm not sure what else to check as it's a very simple piece of config.